webapp/channels/src/components/channel_layout/channel_controller.tsx

```diff
@@ -40,6 +41,8 @@ export default function ChannelController(props: Props) {
 
         return () => {
             document.body.classList.remove(...BODY_CLASS_FOR_CHANNEL);
+            cleanupUserProfilesInterval();
+            cleanupUserStatusesInterval();
```

Comment:
Can you add a comment as to why this is needed here?

Reply:
I'm trying to think if there's a nicer way to do this because it's a bit weird to me that the action starts an interval and this seemingly-unrelated code stops it. It would be nice if the same piece of code could be responsible for both.

What happens if we don't clean these up? It'll just continue to run the interval for the rest of the app's lifecycle? It feels a bit wasteful, but given most of the app is contained within the ChannelController still, we might not need to worry about pausing the interval when you go to a different part of the app

Reply:
Comment added. Thats correct we might not want to pause since most of this is in channel_controller but its always a good idea to do clean ups